components/user_settings/user_settings_advanced.jsx

```diff
@@ -77,11 +77,15 @@ export default class AdvancedSettingsDisplay extends React.Component {
 
         const isSaving = false;
 
-        return {preReleaseFeatures: PreReleaseFeatures,
+        const previewFeaturesEnabled = global.window.mm_config.EnablePreviewFeatures === 'true';
```

Comment:
The [default.json](https://github.com/mattermost/mattermost-server/blob/master/config/default.json#L56) uses ` "EnablePreviewFeatures": true` (with a Boolean value) but this is comparing against a string, and in JavaScript:
```javascript
'true' === true // false
```
We should either compare against a Boolean only or both a Boolean and a string.

Reply:
Unfortunately we do this everywhere because the server returns the client config as a JSON object with string values (an oversight from very early in the project)

We will eventually fix it but no one's gotten around to it yet

Reply:
Oh OK. I didn't realized. Thanks.